### ISSUE TITLE
[feature request] ability to import logic gate representation

### ISSUE THREAD

**Issue by Ponali** [SebLague/Digital-Logic-Sim#211]
it would be nice to be able to import a text representation of logic gates, so it can later be turned into its own chip for abstraction.

in the "Exploring How Computers Work" series made for this, when talking about 7-segment (or 8?) displays, there has been a moment for a program called "Logic Friday". i installed this program, put a truth table and converted it into a logic gate representation.
i can use the diagram to reconstruct it in the simulation, but it will take a lot of time to do so, and the diagram is veery complicated '._.

here's what the text representation looks like to me:
```
F0 = 0;
F1 = A' B C  + A' B' C  + A B C  + A B D' + A' B C' D + B' C' D' + A B' C'  + A B' D';
F2 = A' C' D' + A' B' D + A B C' D + A' B' C  + A B' C'  + A B' D';
F3 = A' B' D + A' B C  + A' B D' + A B C' D + A B' C D + A' B C' D + B' C' D' + A B' C'  + A B' D';
F4 = A' C D' + A B C' D + A B' C D + A' B' C  + A B D' + A' B C' D + B' C' D' + A B' C' ;
F5 = A' C D' + A B C' D + A B' C D + A B C  + A B D' + B' C' D' + A B' D';
F6 = A' B D' + A B' C D + A B C  + A B D' + A' B C' D + B' C' D' + A B' C'  + A B' D';
F7 = A' B D' + A B C' D + A B' C D + A' B' C  + A B C  + A' B C' D + A B' C'  + A B' D';
```

if you're wondering why i'm not copying from the video with the version with Two's Complement, that's because i wanted to make my own version using hexadecimal so i can save some time.

**Comment by BestGiter** [comment]
maybe truth table -> chips

**Comment by MattheDev53** [comment]
I agree 100% with this idea. Maybe it could even have a SuperTuxKart-esque menu for downloading blocks from the internet (or even locally)
